### PR TITLE
Updates tslint configuration

### PR DESCRIPTION
**Pull request by gorkem**
Several changes related to tslint.
1. tslint is now invoked as as a typescript plugin,
and the gulp function related to tslint is removed.
2. updated tslint.json with rules.
3. fixed reported tslint errors.

### CODE REVIEW

**Comment by fbricon** [comment]
@gorkem can you sign-off your commit?

File: tslint.json [hunk 1]
```diff
@@ -1,22 +1,57 @@
...
-		"no-unsafe-finally": true,
-		"new-parens":true,
-		"no-duplicate-imports": true,
-		"deprecation":true,
-		"no-unsafe-any":true,
-		"no-string-throw":true,
-		"no-require-imports": true
-	}
+    "rules": {
+        "encoding": true,
```
Comment (yaohaizh) [inline tslint.json]: Should we enable 
```
"extends": "tslint:recommended"
```

Comment (gorkem) [reply tslint.json]: We could but TBH there are a few rules there that really annoys at least me.

**Comment by fbricon** [comment]
merged as https://github.com/redhat-developer/vscode-java/commit/831f44f4c930b4328622448104b6513a0bbead75

**Pull request closed without merging**